[DSS-285] Update Button story to show one story in sidebar

goodwinchris opened the pull request:
# Description

Update Sage-Button story to show only one default example of our button. For all variants we now add a `<sage-button />` to the page. This limits the number of button stories in the sidebar to 1. 

[Fixes DSS-285](https://kajabi.atlassian.net/browse/DSS-285)

## Type of change

Update to storybook

# How Has This Been Tested?

There should be no testing as the component hasn't been changed, only storybook is updated.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

Commit:
docs(button): update to show only one button story and example instances of  instead

Commit:
docs(button): update button story name/url

ju-Skinner commented on libs/core/src/components/sage-button/stories/sage-button.stories.mdx:
```diff
@@ -43,80 +50,50 @@ When following our usage guidelines, sage-button should be accessible right out
+<Canvas withSource="none">
+  <sage-button>Default</sage-button>
+  <sage-button disabled="true">Disabled</sage-button>
+  <sage-button icon="trashIcon">Icon</sage-button>
```
Maybe I wasn't clear previously, but I was ok with multiple stories, e.g Primary, Secondary, Accent. I felt the slight variants `PrimaryIcon`, and `PrimaryDisabled` was a little too much.

goodwinchris replied:
Gotcha, updates pushed 👍

Commit:
docs(button): update storybook to show a story for each button variant

ju-Skinner committed:
chore(docs): modify the stories structure to separate code from mdx documentation

ju-Skinner committed:
chore(storybook): change blob to use `docs.stories.mdx`

Until we rework the existing stories we have to add `mdx` to the second
blob in the list. Once that work is complete we will remove it from the
regex.

anechol reviewed (approved):
Now that I'm seeing this live, I really love the separation here. Looks super clean and organized, and we can still be explicit while giving the engineers the exact scenario they need. Nice!!

QuintonJason reviewed (approved):
Looks great!
